cg/apps/lims/api.py

```diff
@@ -219,72 +232,110 @@ def update_sample(self, lims_id: str, sex=None, application: str=None,
-            arts = self.get_artifacts(process_type=process_name, samplelimsid=lims_id)
-            if arts:
-                prep_artifact = arts[0]
-                method_number = (
-                        prep_artifact.parent_process.udf.get('Method') or
-                        prep_artifact.parent_process.udf.get('Method Document 1')
-                )
-                method_version = (
-                        prep_artifact.parent_process.udf.get('Version') or
-                        prep_artifact.parent_process.udf.get('Document 1 Version')
-                )
-                break
 
-        if method_number is None:
-            return None
+        step_names_udfs = MASTER_STEPS_UDFS['get_sequencing_method_step']
 
-        method_name = AM_METHODS.get(method_number)
-        return f"{method_number}:{method_version} - {method_name}"
+        return self._get_methods(step_names_udfs, lims_id)
 
     def get_delivery_method(self, lims_id: str) -> str:
         """Get the delivery method."""
-        process_name = 'CG002 - Delivery'
-        arts = self.get_artifacts(process_type=process_name, samplelimsid=lims_id)
-        if arts:
-            prep_artifact = arts[0]
-            method_number = prep_artifact.parent_process.udf.get('Method Document')
-            method_version = prep_artifact.parent_process.udf.get('Method Version')
-        else:
-            return None
 
-        method_name = AM_METHODS.get(method_number)
-        return f"{method_number}:{method_version} - {method_name}"
+        step_names_udfs = MASTER_STEPS_UDFS['get_delivery_method_step']
+
+        return self._get_methods(step_names_udfs, lims_id)
 
     def get_processing_time(self, lims_id):
         received_at = self.get_received_date(lims_id)
         delivery_date = self.get_delivery_date(lims_id)
         if received_at and delivery_date:
             return delivery_date - received_at
+
+    @staticmethod
+    def _sort_by_date_run(sort_list: list):
+        """
+        Sort list of tuples by parent process attribute date_run in descending order.
+
+        Parameters:
+            sort_list (list): a list of tuples in the format (date_run, elem1, elem2, ...)
+
+        Returns:
+            sorted list of tuples
+        """
+        return sorted(sort_list, key=lambda sort_tuple: sort_tuple[0], reverse=True)
+
+    def _most_recent_date(self, dates: list):
+        """
+        Gets the most recent date from a list of dates sorted by date_run
+
+        Parameters:
+            dates (list): a list of tuples in the format (date_run, date), sorted by date_run
+            descending
+
+        Returns:
+            The date in the first tuple in dates
+        """
+        sorted_dates = self._sort_by_date_run(dates)
+        date_run_index = 0
+        date_index = 1
+
+        return sorted_dates[date_run_index][date_index] if dates else None
+
+    def _get_methods(self, step_names_udfs, lims_id):
```

Comment:
I see that we get lint errors on this method since we it has more than 15 local variables, do you see how we could reduce that number?

Reply:
I'm going to look at that. I'm looking at other linting stuff too btw.